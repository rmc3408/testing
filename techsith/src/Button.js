import React from 'react'

const Button = ({label}) => {
    return (
        <div className='btn' data-testid='button'>Hi {label}</div>
    )
}

export default Button
