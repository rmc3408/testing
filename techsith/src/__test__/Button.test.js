import React from 'react';
import ReactDOM from 'react-dom';
import Button from '../Button';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';



afterEach(cleanup);

it('render withot crash', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Button></Button>, div);
})

test('render button label', () => {
    const { getByTestId } = render(<Button label='click me'></Button>);
    expect(getByTestId('button')).toHaveTextContent('click me');
    
})

test('render button label 2', () => {
    const { getByTestId } = render(<Button label='save'></Button>);
    expect(getByTestId('button')).toHaveTextContent('save')
})

test('snapshot', () => {
    const myTree = renderer.create(<Button label='snap'></Button>).toJSON();
    expect(myTree).toMatchSnapshot();
})


export default Button
