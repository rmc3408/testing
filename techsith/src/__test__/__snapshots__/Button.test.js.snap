// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="btn"
  data-testid="button"
>
  Hi 
  snap
</div>
`;
